\chapwithtoc{Discussion}
\label{discussion}

The length and amino acid composition of non-domain regions can be crucial for the
regulation of multi-domain proteins' activity, including protein
kinases~\cite{gogl2019disordered, vigil2004conformational}. % PRIDEJ DALSI ZDROJE
However, to our knowledge, the influence of the linkers' composition on the overall
protein function has not been described \emph{in general} yet.
This thesis tried to address this issue by selecting a dataset of evolutionarily related
multi-domain proteins containing a protein kinase domain, clustering these molecules
based on the physicochemical attributes of their non-domain regions, and by identifying
Gene Ontology terms and Enzyme Commission numbers specific to the detected clusters of
proteins with various architectures.

Even though it was possible to divide proteins from the studied dataset into three groups
based on the clusters visible from the UMAP dimensionality reduction of the 4-dimensional
vectors of the linkers' normalized physicochemical characters, no frequent functional
annotation term could characterize the defined clusters.
There may be several reasons for the lack of success of the designed method.
For example, there may really not be any overall influence of the non-domain regions'
composition on the function of two-domain protein kinases.
However, based on the literature search presented above, this proposition seems rather
impobable~\cite{winkler1977tomato, van1997linker, ikebe1998hinge, robinson1998optimizing,
rice1999structural, gokhale2000role, case2000role, pufall2002autoinhibitory,
khalil2008kinesin, hariharan2009insights, smock2010interdomain, liu2010molecular,
shastry2010neck, ma2011dynamic, cyrus2011impact, kalodimos2011nmr, pohane2015modulation,
klement2015effect, rozycki2017length, jakubec2018widespread, gogl2019disordered}.

The narrow dataset may be a significant problem.
In a sample of large size, more information is available, on the other hand, false
presumptions may be concluded from a small sample generating the
observations~\cite{tanaka1987big, hua2005optimal}.
In the studied dataset, 32 different architectures are present within the 117 molecules.
Proteins with the same architecture are frequently related, both evolutionarily and
functionally~\cite{vogel2004structure, hegyi2001annotation, bashton2002geometry}, and the
linkers may introduce new structural features~\cite{papaleo2016role}, thus enhancing
functionality.
With 3.65625 proteins per architecture on average, no feasible resolution can be
concluded on how the composition of the linkers affects the behavior of these molecules.

Furthermore, both function annotation services, Gene Ontology and Enzyme Commission, have
drawbacks regarding their completeness and homogeneity.
For instance, \citet{gaudet2017gene} claim, that the GO is biased, and unevenly
incomplete.
The Gene Ontology database is a reflection of primary literature~\cite{gene2004gene},
therefore less studied areas are inadequately represented in the GO.
On the other hand, more comprehensively covered parts of the GO are not flawless as well,
as they can provide contradictory information, which can be caused for example by
differences in experimental conditions of comparable
research:~\cite{hass2004response, mason2005multiple}.

Within the studied 117 human two-domain protein kinases, total 1,982 GO annotations were
fetched from the \texttt{uniprot\_reference\_proteomes.dat} file, peaking at 69 terms
assigned to a single protein, namely the \texttt{AAPK1\_HUMAN} kinase.
These include even GO terms from the same hierarchy.
For example, it is possible to reach \texttt{GO:0008610; lipid biosynthetic process} by
recursively applying the transitive relationship ``\texttt{is\_a}'' on
\texttt{GO:0045542; positive regulation of cholesterol biosynthetic process}, which are
both explicitly mentioned in the \texttt{uniprot\_reference\_proteomes.dat} file.
Furthermore, the same protein is classified as \texttt{2.7.11.1 non-specific
serine/threonine protein kinase}, however, at the same time, has the specific EC number
\texttt{2.7.11.26}, \texttt{2.7.11.27}, and \texttt{2.7.11.31} assigned to it.

To be assigned an Enzyme Commission number, there must be a direct experimental evidence
that the proposed enzyme actually catalyses the claimed reaction~\cite{mcdonald2014fifty}.
60 protein from the studied dataset were described as \texttt{2.7.11.1 non-specific
serine/threonine protein kinase}, meaning that these kinases are either non-specific,
or their specificity has not been analyzed to
date\footnote{\url{https://www.brenda-enzymes.org/enzyme.php?ecno=2.7.11.1}}.
Neither of these possibilities comprise good news for the goal of this thesis.

The strict nature of EC, as well as the excessiveness of the GO combined with insufficient
amount of research could be overcome by measuring similarities of the GO
terms~\cite{li2010effectively, zhao2018gogo} within the protein clusters, instead of
enforcing their absolute exclusion.
Nevertheless, a larger dataset is required to produce a significant outcome.

% \chapter{Results and discussion}
%
% You should have a separate chapter for presenting your results (generated by the stuff described previously, in our case in \cref{chap:math}). Remember that your work needs to be validated rigorously, and no one will believe you if you just say that `it worked well for you'.
%
% Instead, try some of the following:
% \begin{itemize}
% \item State a hypothesis and prove it statistically
% \item Show plots with measurements that you did to prove your results (e.g. speedup). Use either \texttt{R} and \texttt{ggplot}, or Python with \texttt{matplotlib} to generate the plots.\footnote{Honestly, the plots from \texttt{ggplot} look \underline{much} better.} Save them as PDF to avoid printing pixels (as in \cref{fig:f}).
% \item Compare with other similar software/theses/authors/results, if possible
% \item Show example source code (e.g. for demonstrating how easily your results can be used)
% \item Include a `toy problem' for demonstrating the basic functionality of your approach and detail all important properties and results on that
% \item Include clear pictures of `inputs' and `outputs' of all your algorithms, if applicable
% \end{itemize}
%
% \begin{figure}
% \centering
% \includegraphics[width=.6\linewidth]{img/ukazka-obr01.pdf}
% \caption{This caption is a friendly reminder to never insert figures ``in text,'' without a floating environment, unless explicitly needed for maintaining the text flow (e.g., the figure is small and developing with the text, like some of the centered equations, as in \cref{thm:y}). All figures \emph{must} be referenced by number from the text (so that the reader can find them when he reads the text) and properly captioned (so that the reader can interpret the figure even if he looks at it before reading the text --- reviewers love to do that).}
% \label{fig:f}
% \end{figure}
%
% It is sometimes convenient (even recommended by some journals, including Cell) to name the results sub-sections so that they state what exactly has been achieved. Examples follow.
%
% \section{SuperProgram is faster than OldAlgorithm}
% \subsection{Scalability estimation}
% \subsection{Precision of the results}
% \section{Weird theorem is proven by induction}
% \section{Amount of code reduced by CodeRedTool}
% \subsection{Example}
% \subsection{Performance on real codebases}
% \section{\sloppy NeuroticHelper improves neural network learning}
%
% \section{What is a discussion?}
% After you present the results and show that your contribution works, it is important to \emph{interpret} them, showing what they mean for the more general public.
%
% Separate discussion sections are common in life sciences where ambiguity is common and intuition is sometimes the only thing that the authors have; exact sciences and mathematicians do not use them as often. Despite of that, it is nice to precisely set your output into the existing environment, answering:
% \begin{itemize}
% \item What is the potential application of the result?
% \item Does the result solve a problem that other people encountered?
% \item Did the results point to any new (surprising) facts?
% \item Why is the result important for your future work (or work of anyone other)?
% \item Can the results be used to replace (and improve) anything that is used currently?
% \end{itemize}
%
% If you do not know the answers, you may want to ask the supervisor. Also, do not worry if the discussion section is half-empty or completely pointless; you may remove it completely without much consequence. It is just a bachelor thesis, not a world-saving avenger thesis.
