
\chapwithtoc{Introduction}

Proteins are amino acid chains, polypeptides, that serve a variety of functions within
living cells, including structural support and movement, interactions with cell's
environment, and biochemical reaction catalysis~\cite{alberts2018molecular}.
To function properly, proteins have to fold into their native conformation, and as
demonstrated by \citet{anfinsen1961kinetics} on bovine pancreatic ribonuclease, the
information for correct folding is contained in the amino acid sequence itself.

Different sequences fold into different three-dimensional conformations.
Certain local regions form secondary structures, such as $\alpha$ helices and $\beta$
sheets, and these locally ordered regions associate to form the whole protein, or in the
case of some larger proteins, to form folding units~\cite{levitt1975computer}.
If the folding units would be stable, given that the polypeptide chain connecting them
to the rest of the protein molecule were to be cleaved, these subassemblies can be
referred to as \emph{domains}~\cite{goldberg1969tertiary, levitt1975computer}.
Nevertheless, today's definition requires that these subassemblies also concievably
function in isolation, and members of the same \emph{domain family} tend to possess an
ancient evolutionary relationship and often a similar function~\cite{ponting2002natural}.

Many bioinformatical tools are available online to perform domain identification within a
protein sequence.
The \emph{SCOP} database organises proteins of known three-dimensional structures according to their evolutionary and structural relationship~\cite{murzin1995scop}.
It classifies non-redundant protein domains and defines them on two main levels of SCOP
classification, family and superfamily.
The designation of proteins in SCOP has been constructed mainly
manually~\cite{andreeva2020scop}.

Other tools tend to implement semimanual approaches, typically including a
\emph{profile hidden Markov Model} (HMM), a powerful probabilistic method describing the
sequence conservation in a family~\cite{krogh1994hidden, eddy1996hidden}.
\emph{Pfam}~\cite{sonnhammer1997pfam} is such an example.
Each protein family in Pfam database consists of a seed alignment forming the basis to
build a HMM-based profile~\cite{el2019pfam} by engaging the HMMER
software~\cite{finn2010pfam, finn2011hmmer}.
Another protein structure classification database using HMMs to scan protein sequences
against it is called \emph{CATH}~\cite{dawson2017cath}.
CATH clusters proteins on four main levels, class (C), architecture (A), topology (T), and
homologous superfamily (H)~\cite{orengo1997cath}.
CATH, Pfam, and many more, are then integrated in a general resource for protein families,
domains, and functional sites, called InterPro~\cite{finn2017interpro}.
The authors aim to create a non-redundant characterization and the software package
InterProScan provides an interface to functionally classify novel nucleotide or protein
sequences~\cite{zdobnov2001interproscan}.
By uniting member databases, InterPro exploits their individual strengths, thus
significantly contributing in the troublesome effort of automatic
annotation~\cite{apweiler2000interpro}.



%Introduction should answer the following questions, ideally in this order:
%\begin{enumerate}
%\item What is the nature of the problem the thesis is addressing?
%\item What is the common approach for solving that problem now?
%\item How this thesis approaches the problem?
%\item What are the results? Did something improve?
%\item What can the reader expect in the individual chapters of the thesis?
%\end{enumerate}

%Expected length of the introduction is between 1--4 pages. Longer introductions may require sub-sectioning with appropriate headings --- use \texttt{\textbackslash{}section*} to avoid numbering (with section names like `Motivation' and `Related work'), but try to avoid lengthy discussion of anything specific. Any ``real science'' (definitions, theorems, methods, data) should go into other chapters.
%\todo{You may notice that this paragraph briefly shows different ``types'' of `quotes' in TeX, and the usage difference between a hyphen (-), en-dash (--) and em-dash (---).}

%It is very advisable to skim through a book about scientific English writing before starting the thesis. I can recommend `\citetitle{glasman2010science}' by \citet{glasman2010science}.
